src/OptimalTransport.jl

```diff
@@ -672,4 +675,162 @@ function quadreg(mu, nu, C, ϵ; θ=0.1, tol=1e-5, maxiter=50, κ=0.5, δ=1e-5)
     return sparse(γ')
 end
 
+## 1-Dimensional Optimal Transport Functions
+
+"""
+optimal_transport_cost(c,mu::ContinuousUnivariateDistribution,nu::UnivariateDistribution; plan=nothing)
+
+Calculates the optimal transport cost between μ to ν, where
+they are 1-Dimensional distributions and the cost
+function is of the form ``c(x,y) = h(|x-y|)`` such that
+``h`` is a convex function.
+
+Parameters:\n
+c: The cost function, which should be of the form ``c(x,y) = h(abs(x-y))``
+where `h` is a convex function.
+mu: 1-D distribution (e.g. `mu = Distributions.Normal(0,1)`)\n
+nu: 1-D distribution (e.g. `nu = Distributions.Normal(0,1)`)\n
+plan: Optional parameter in case the optimal transport plan is already known. Providing
+it can speed up calculations.
+"""
+function optimal_transport_cost(
+    c, μ::ContinuousUnivariateDistribution, ν::UnivariateDistribution; plan=nothing
+)
+    if plan === nothing
+        g(μ, ν, x) = c(quantile(μ, x), quantile(ν, x))
+        f(x) = g(μ, ν, x)
+        return quadgk(f, 0, 1)[1]
+    else
+        cost = quadgk(0, 1) do q
+            x = quantile(μ, q)
+            return c(x, plan(x))
+        end
+        return cost[1]
+    end
```

Comment:
There is no need to define `g` and in particular `f` explicitly here:
```suggestion
    cost, _ = if plan === nothing
        quadgk(0, 1) do q
            return c(quantile(μ, q), quantile(ν, q))
        end
    else
        quadgk(0, 1) do q
            x = quantile(μ, q)
            return c(x, plan(x))
        end
    end
    return cost
```

Reply:
Should I do the same for the `x = quantile (\mu,q)`?

Reply:
No, that's fine. It's not about avoiding all variable names but reducing unnecessary function definition since it creates overhead, both mentally and for the compiler.